CMakeLists.txt

```diff
@@ -1,91 +1,195 @@
+	src/generated/ldb_battlecommands.cpp
+	src/generated/ldb_battleranimation.cpp
+	src/generated/ldb_battleranimationdata.cpp
+	src/generated/ldb_battleranimationextension.cpp
+	src/generated/ldb_chipset.cpp
+	src/generated/ldb_class.cpp
+	src/generated/ldb_commonevent.cpp
+	src/generated/ldb_database.cpp
+	src/generated/ldb_enemyaction.cpp
+	src/generated/ldb_enemy.cpp
+	src/generated/ldb_itemanimation.cpp
+	src/generated/ldb_item.cpp
+	src/generated/ldb_learning.cpp
+	src/generated/ldb_music.cpp
+	src/generated/ldb_skill.cpp
+	src/generated/ldb_sound.cpp
+	src/generated/ldb_state.cpp
+	src/generated/ldb_switch.cpp
+	src/generated/ldb_system.cpp
+	src/generated/ldb_terms.cpp
+	src/generated/ldb_terrain.cpp
+	src/generated/ldb_terrain_flags.cpp
+	src/generated/ldb_testbattler.cpp
+	src/generated/ldb_troop.cpp
+	src/generated/ldb_troopmember.cpp
+	src/generated/ldb_trooppagecondition.cpp
+	src/generated/ldb_trooppagecondition_flags.cpp
+	src/generated/ldb_trooppage.cpp
+	src/generated/ldb_variable.cpp
+	src/generated/lmt_encounter.cpp
+	src/generated/lmt_mapinfo.cpp
+	src/generated/lmt_start.cpp
+	src/generated/lmu_event.cpp
+	src/generated/lmu_eventpagecondition.cpp
+	src/generated/lmu_eventpagecondition_flags.cpp
+	src/generated/lmu_eventpage.cpp
+	src/generated/lmu_map.cpp
+	src/generated/lmu_moveroute.cpp
+	src/generated/lsd_saveactor.cpp
+	src/generated/lsd_savecommonevent.cpp
+	src/generated/lsd_save.cpp
+	src/generated/lsd_saveeventcommands.cpp
+	src/generated/lsd_saveeventdata.cpp
+	src/generated/lsd_saveinventory.cpp
+	src/generated/lsd_savemapevent.cpp
+	src/generated/lsd_savemapinfo.cpp
+	src/generated/lsd_savepartylocation.cpp
+	src/generated/lsd_savepicture.cpp
+	src/generated/lsd_savepicture_flags.cpp
+	src/generated/lsd_savescreen.cpp
+	src/generated/lsd_savesystem.cpp
+	src/generated/lsd_savetarget.cpp
+	src/generated/lsd_savetitle.cpp
+	src/generated/lsd_savevehiclelocation.cpp
+	src/generated/rpg_chipset.cpp
+	src/generated/rpg_mapinfo.cpp
+)
+
+# C++11 is required
+set_property(TARGET lcf PROPERTY CXX_STANDARD 11)
+set_property(TARGET lcf PROPERTY CXX_STANDARD_REQUIRED ON)
+
+# includes
+target_include_directories(
+	lcf PUBLIC
+	${CMAKE_CURRENT_SOURCE_DIR}/src
+	${CMAKE_CURRENT_SOURCE_DIR}/src/generated
+)
+
 # endianess checking
 if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
-  include(TestBigEndian)
-  test_big_endian(WORDS_BIGENDIAN)
-  if(WORDS_BIGENDIAN)
-    add_definitions(-DWORDS_BIGENDIAN)
-  endif()
+	include(TestBigEndian)
+	test_big_endian(WORDS_BIGENDIAN)
+	if(WORDS_BIGENDIAN)
+		target_compile_definitions(lcf PRIVATE WORDS_BIGENDIAN=1)
+	endif()
 endif()
 
-# C++11 support is needed
-include(CheckCXXCompilerFlag)
-CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-if(COMPILER_SUPPORTS_CXX11)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+# icu
+if(DISABLE_ICU)
+	find_package(Iconv REQUIRED)
+	target_link_libraries(lcf Iconv::Iconv)
+	if(NOT Iconv_IS_BUILT_IN)
+		list(APPEND LIBLCF_DEPS "iconv")
```

Comment:
not sure this does the right thing. `requires` and `requires.private` are for dependencies that `pkg-config` is able to resolve. however, gnu libiconv does not ship a pkg-config file (AFAIR) and a system library used on some unknown platform likely neither.